src/Components/SelectedProduct.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect} from "react";
+import axios from "axios";
+import { useDispatch, useSelector } from "react-redux";
+import { selectedProduct } from "../redux/actions/productActions";
```

Comment:
avoid giving conflicting name to the action. u can rename as selectedProductAction.
import { selectedProduct as selectedProductAction } from "../redux/actions/productActions"